src/zjs_console.c

```diff
@@ -34,48 +34,99 @@ int is_int(jerry_value_t val) {
 {
-    int i;
-    for (i = 0; i < argc; ++i) {
-        if (jerry_value_is_string(argv[i])) {
-            jerry_size_t jlen = jerry_get_string_size(argv[i]);
-            if (jlen > MAX_STR_LENGTH) {
-                ERR_PRINT("console string is too long, max length=%u\n", MAX_STR_LENGTH);
-                return ZJS_UNDEFINED;
+    if (jerry_value_is_array(value)) {
+        uint32_t len = jerry_get_array_length(value);
+        if (deep) {
+            fprintf(out, "[");
+            for (int i = 0; i < len; i++) {
+                if (i) {
+                    fprintf(out, ", ");
+                }
+                jerry_value_t element = jerry_get_property_by_index(value, i);
+                print_value(element, out, false, true);
             }
-            char buffer[jlen + 1];
-            int wlen = jerry_string_to_char_buffer(argv[i], (jerry_char_t *)buffer, jlen);
-            buffer[wlen] = '\0';
-            fprintf(out, "%s ", buffer);
-        } else if (jerry_value_is_number(argv[i])) {
-            int type = is_int(argv[i]);
-            if (type == IS_NUMBER) {
-                double num = jerry_get_number_value(argv[i]);
-                fprintf(out, "%f ", num);
-            } else if (type == IS_UINT) {
-                uint32_t num = (uint32_t)jerry_get_number_value(argv[i]);
-                fprintf(out, "%lu ", num);
-            } else if (type == IS_INT) {
-                int32_t num = (int32_t)jerry_get_number_value(argv[i]);
-                // Linux and Zephyr print int32_t's differently if %li is used
+            fprintf(out, "]");
+        }
+        else {
+            fprintf(out, "[Array - length %lu]", len);
+        }
+    }
+    else if (jerry_value_is_boolean(value)) {
+        uint8_t val = jerry_get_boolean_value(value);
+        fprintf(out, (val) ? "true" : "false");
+    }
+    else if (jerry_value_is_function(value)) {
+        fprintf(out, "[Function]");
+    }
+    else if (jerry_value_is_number(value)) {
+        int type = is_int(value);
+        if (type == IS_NUMBER) {
+            double num = jerry_get_number_value(value);
+            fprintf(out, "%f", num);
+        } else if (type == IS_UINT) {
+            uint32_t num = (uint32_t)jerry_get_number_value(value);
+            fprintf(out, "%lu", num);
+        } else if (type == IS_INT) {
+            int32_t num = (int32_t)jerry_get_number_value(value);
+            // Linux and Zephyr print int32_t's differently if %li is used
 #ifdef ZJS_LINUX_BUILD
-                fprintf(out, "%i ", num);
+            fprintf(out, "%i", num);
 #else
-                fprintf(out, "%li ", num);
+            fprintf(out, "%li", num);
 #endif
+        }
+    }
+    else if (jerry_value_is_null(value)) {
+        fprintf(out, "null");
+    }
+    // NOTE: important that checks for function and array were above this
+    else if (jerry_value_is_object(value)) {
+        fprintf(out, "[Object]");
+    }
+    else if (jerry_value_is_string(value)) {
```

Comment:
Do you want me to fix this? I am confused... Isn't this your PR? Or did you mean to comment on my other floating point printing PR (https://github.com/01org/zephyr.js/pull/466).

Or were you just commenting saying that you were changing the style of what I originally wrote. Changing it is fine with me. I agree it makes the code more readable, I am just very used to doing it the way I did it out of habit.
Just making sure, do we want ALL if-else blocks like this? or only one line blocks to break it up?